### PR TITLE
Enhancement: Use `coverallsapp/github-action` to report code coverage

### PR DESCRIPTION
**Pull request by localheinz**
This pull request

- [x] uses [`coverallsapp/github-action`](https://github.com/marketplace/actions/coveralls-github-action) to report code coverage

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -135,12 +135,12 @@ jobs:
           make ci-test
 
       - name: Coveralls
-        run: vendor/bin/php-coveralls -v --coverage_clover='clover.xml' --json_path='coveralls-upload.json'
-        if: env.COVERALLS_REPO_TOKEN && matrix.php-version == '8.1' && matrix.env == 'AUTOLOAD=1'
```
Comment (localheinz) [inline .github/workflows/ci.yml:139]: We can probably remove the `COVERALLS_REPO_TOKEN` environment variable (if present) from https://github.com/stripe/stripe-php/settings/variables/actions.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -135,12 +135,12 @@ jobs:
...
-        env:
-          COVERALLS_RUN_LOCALLY: 1
-          COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_REPO_TOKEN }}
-          COVERALLS_FLAG_NAME: php-${{ matrix.php-version }}-${{ matrix.env }}
+        if: matrix.php-version == '8.1' && matrix.env == 'AUTOLOAD=1'
+        uses: coverallsapp/github-action@v2
+        with:
+          files: clover.xml
+          flag-name: php-${{ matrix.php-version }}-${{ matrix.env }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (localheinz) [inline .github/workflows/ci.yml:143]: See https://github.com/marketplace/actions/coveralls-github-action#inputs.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -130,12 +130,12 @@ jobs:
           make ci-test
 
       - name: Coveralls
-        run: vendor/bin/php-coveralls -v --coverage_clover='clover.xml' --json_path='coveralls-upload.json'
-        if: env.COVERALLS_REPO_TOKEN && matrix.php-version == '8.2' && matrix.env == 'AUTOLOAD=1'
-        env:
-          COVERALLS_RUN_LOCALLY: 1
-          COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_REPO_TOKEN }}
```
Comment (localheinz) [inline .github/workflows/ci.yml:137]: We can probably remove the `COVERALLS_REPO_TOKEN` environment variable (if present) from https://github.com/stripe/stripe-php/settings/variables/actions.

**Review by pakrym-stripe — APPROVED** [review]
This is great!

**Pull request merged**